common/src/main/java/io/druid/math/expr/ExprEval.java

```diff
@@ -242,25 +242,37 @@ public final ExprType type()
     @Override
     public final boolean isNull()
     {
-      return Strings.isNullOrEmpty(value);
+      return value == null;
     }
 
     @Override
     public final int asInt()
     {
-      return Integer.parseInt(value);
+      if (value == null) {
+        return 0;
+      }
+
+      final Integer theInt = Ints.tryParse(value);
+      return theInt == null ? 0 : theInt;
```

Comment:
I thought it would be more useful to treat unparseable strings as zeroes rather than to throw errors. I don't want one bad string in a column to make the entire query fail. It's similar to the behavior of Druid's "numeric" comparator on string columns, where unparseable numbers as less than all other numbers, not as errors.